### DISCUSSION TITLE
Multi-dimensional vs. Jagged Arrays

### DISCUSSION THREAD

**Discussion by nooelan** [Handlebars-Net/Handlebars.Net#450] [Ideas]
My model contains data that are multi-dimensional, and my first attempt was to implement this as a regular C# multidimensional array similar to this:

`object[,] matrix3x3 = new object[3,3];`

When trying to access this type of object from a template, Handlebars.Net threw an exception complaining it could not convert from object[ , ] to object[].

Then I changed my model implementation to jagged arrays (similar to this):

```
object[][] matrix3x3 = new object[3]; 
for (row = 0; row < 3; ++row)
    matrix3x3[row] = new object[3];
```

This works just fine with Handlebars.Net.

Question: Is the lack of support for regular multidimensional arrays intentional, or is it a missing feature or bug?

As regular, multidimensional arrays are more efficient, at least for 2D arrays according to C# guidelines, I think it is a good idea to add support for it. It will also help avoid having to change the implementation of an already existing model.

**Comment by oformaniuk** [comment]
Hello @nooelan 
Hamdlebars.Net is trying to keep feature parity with Handlebarsjs. Unfortunately multidimensional arrays are not supported in JavaScript and as a result would be never supported in Handlebarsjs.

You can provide your custom `ObjectDescriptor` that supports multidimensional  arrays. However, you'd be only able to virtually project values to jagged array due to syntax limitations. Then usage will be something like this:
```handlebars
Value at [2,2]: {{array.2.2}}
```

Note: `[1]` is not an indexer syntax. It's rather way to specify non-valid entries (like words separated by `space`) in a correct way.
